terraform/aws/modules/efs/efs.tf

```diff
@@ -0,0 +1,24 @@
+/*This will create EFS in bursting mode.The mode can be changed as per the requirement*/
+resource "aws_efs_file_system" "elastic_file_system" {
+  creation_token = var.creation_token
+
+  lifecycle_policy {
+    transition_to_ia = "AFTER_30_DAYS"
+  }
+  encrypted        = var.encryption_set
+  kms_key_id       = aws_kms_key.kms_key_efs.arn
+  performance_mode = var.performance_mode
+  throughput_mode  = var.throughput_mode
+}
+
+# Define the EFS mount target (optional)
+resource "aws_efs_mount_target" "efs_mount_target" {
+  file_system_id  = aws_efs_file_system.elastic_file_system.id
```

Comment:
please add a count logic. Only create the mount target, if the subnet is provided. Additionally, you can change subnet to be a list, that way for each subnet provided you can create a new mount target. This is useful in cases of multi AZ.

Reply:
@qburst-praven This is done.